ngraph/core/include/openvino/core/preprocess/input_info.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/preprocess/input_tensor_info.hpp"
+#include "openvino/core/preprocess/preprocess_steps.hpp"
+
+namespace ov {
+namespace preprocess {
+
+class InputInfoImpl;
+
+/// \brief Class holding preprocessing information for one input
+/// API has Builder-like style to allow chaining calls in client's code, like
+/// \code{.cpp}
+/// auto proc = PrePostProcessor().input(InputInfo().tensor(...).preprocess(...);
+/// \endcode
+class OPENVINO_API InputInfo final {
+    std::unique_ptr<InputInfoImpl> m_impl;
+    friend class PrePostProcessor;
+
+public:
+    /// \brief Empty constructor. Should be used only if network will have only one input
+    InputInfo();
+
+    /// \brief Information about info for particular input index of model
+    ///
+    /// \param input_index Index to address specified input parameter of model
+    InputInfo(size_t input_index);
```

Comment:
we will add tensor name support?

Reply:
Yes, but later when corresponding API is available. This is just a first part (even without Layout) which is ready for commit and can be tested